packages/typegpu/src/data/array.ts

```diff
@@ -22,34 +22,100 @@ import { getCustomAlignment } from './attributes';
 // Public API
 // ----------
 
+/**
+ * Array schema constructed via `d.arrayOf` function.
+ *
+ * Responsible for handling reading and writing array values
+ * between binary and JS representation. Takes into account
+ * the `byteAlignment` requirement of its elementType.
+ */
 export interface TgpuArray<TElement extends AnyTgpuData>
   extends TgpuData<Unwrap<TElement>[]> {
   readonly elementType: TElement;
   readonly elementCount: number;
 }
 
+/**
+ * Creates an array schema that can be used to construct gpu buffers.
+ * Describes arrays with fixed-size length, storing elements of the same type.
+ *
+ * @example
+ * const LENGTH = 3;
+ * const array = d.arrayOf(d.u32, LENGTH);
+ *
+ * @param elementType The type of elements in the array.
+ * @param count The number of elements in the array.
+ */
 export const arrayOf = <TElement extends AnyTgpuData>(
   elementType: TElement,
   count: number,
 ): TgpuArray<TElement> => new TgpuArrayImpl(elementType, count);
 
+/**
+ * Array schema constructed via `d.looseArrayOf` function.
+ *
+ * Useful for defining tgpu vertex buffers.
+ * Elements in the schema are not aligned in respect to their `byteAlignment`,
+ * unless they are explicitly decorated with the custom align attribute
+ * via `d.align` function.
+ */
 export interface TgpuLooseArray<TElement extends AnyTgpuData | AnyTgpuLooseData>
   extends TgpuLooseData<Unwrap<TElement>[]> {
   readonly elementType: TElement;
   readonly elementCount: number;
 }
 
+/**
+ * Creates an array schema that can be used to construct tgpu vertex buffers.
+ * Describes arrays with fixed-size length, storing elements of the same type.
+ *
+ * Elements in the schema are not aligned in respect to their `byteAlignment`,
+ * unless they are explicitly decorated with the custom align attribute
+ * via `d.align` function.
+ *
+ * @example
+ * const looseArray = d.looseArrayOf(d.u32, 3);
```

Comment:
Tiny nitpick, a loose array of `u32` would be essentially the same as a regular array of `u32`, so maybe a `vec3f` would be a better example?
```suggestion
 * const looseArray = d.looseArrayOf(d.vec3f, 3); // packed array of vec3f
```